### PR TITLE
Put in signal handlers for various signals to all executables

### PR DESCRIPTION
**Pull request by jbakosi**
This is so we have a chance at debugging asynclogic (and other) bugs happening randomly. Most importantly, catching `SIGSEGV`, so we hopefully get a trace for a segfault.

Note also, that the default behavior of attempting to generate a trace is no changed from only in DEBUG to independent of the `CMAKE_BUILD_TYPE`.

---
This change is [Reviewable](https://reviewable.io/reviews/quinoacomputing/quinoa/217)

### CODE REVIEW

File: src/Base/Exception.C [hunk 1]
```diff
@@ -96,14 +92,14 @@ Exception::saveTrace() noexcept
 //!   Requires stdio.h, execinfo.h.
 // *****************************************************************************
 {
+#ifndef HOST_OS_ALPINE
```
Comment (junghans) [inline src/Base/Exception.C]: I would really like to see a comment why ALPINE doesn't have it....

Comment (jbakosi) [reply src/Base/Exception.C]: It does, but segfaults on calling `backtrace()`, and I have not clue how to fix it. I tried playing with the buffer length it takes its 2nd argument, as well as putting the buffer in global scope, making it static, etc. I just cannot fix it. So on alpine we cannot call it.

Comment (jbakosi) [reply src/Base/Exception.C]: Also, https://www.freshports.org/devel/libexecinfo says it's quick'n'dirty, so it is probably broken.

Comment (junghans) [reply src/Base/Exception.C]: Still a comment about alpine OS would be nice

Comment (junghans) [reply src/Base/Exception.C]: I know the commit message has it, but I mean right here in the code.

Comment (jbakosi) [reply src/Base/Exception.C]: Ah, okay, I see. I will put in a comment there then. Thx.

**Pull request closed without merging**